src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,135 @@
     public String getString() {
         return null;
     }
 
     @Override
     public void setString(String string) {
+        this._string = string;
     }
 
+    // this method returns the number of words in the string passed in
     @Override
-    public int count() {
-        return 0;
+    public int count()
+    {
+        int count = 1;
+        String word = _string;
+        if(word.length() == 0)              // if no words are passed in
+        {
+            count = 0;
+        }
+        else
+        {
+            for (int i = 0; i < word.length() - 1; i++)
+            {
+                if (word.charAt(i) == ' ')      // use spaces to split the string into words
+                {
+                    if(i < word.length() - 2)   // logic to keep us from looping out of bounds
+                    {
+                        if(word.charAt(i + 1) != ' ')       // if there are extra spaces ignore them
+                        {
+                            count++;
+                        }
+                    }
+                }
+            }
+        }
+        return count;
     }
 
+    // method to remove every nth character from string
     @Override
-    public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+    public String removeNthCharacter(int n, boolean maintainSpacing)
+    {
+        int index;
+        int nthValue = n;
+        int indexDeleted = 0;
+        String word = _string;
+        StringBuilder newWord = new StringBuilder(word);                // using StringBuilder to manipulate string
+        if(n > word.length() || n <= 0 || word == null)                 // if invalid input, throw exception
+        {
+            throw new IllegalArgumentException("Invalid Input");
+        }
+        for(int i = 0; i < word.length(); i++)                  // loop through entire string
+        {
+            if(i == n - 1)                                      // if at nth place
+            {
+                if(maintainSpacing == true)                     // depending on if spacing is maintained
+                {
+                    newWord.replace(i, i + 1, " ");    // either replace letter with " "
+                }
+                else
+                {
+                    newWord.deleteCharAt(i - indexDeleted);     // or replace letter
+                }
+                indexDeleted++;                             // track changing string length
+                n += nthValue;                              // logic to track every nth value
+            }
+        }
+        return newWord.toString();
     }
 
+    // method to return substrings
     @Override
-    public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+    public String[] getSubStrings(int startWord, int endWord)
+    {
+        int start = 0;
+        String word = _string;
+        String[] words = word.split(" ");                     // split words using spaces
```

Comment:
There is an edge case here when there are two spaces in a row. If an empty string is not considered a valid substring, then we will get an unexpected result for cases like string = "I  like   tuna" and getSubStrings(0,2) which will return ["I","","like"]

Reply:
Ah missed that one, just updated.